central/processbaseline/service/service_impl.go

```diff
@@ -88,14 +93,41 @@ func (s *serviceImpl) GetProcessBaseline(ctx context.Context, request *v1.GetPro
 	}
 	baseline, exists, err := s.dataStore.GetProcessBaseline(ctx, request.GetKey())
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(errox.NotFound, "No process baseline with key %+v found", request.GetKey())
 	}
 	if !exists {
 		return nil, errors.Wrapf(errox.NotFound, "No process baseline with key %+v found", request.GetKey())
 	}
 	return baseline, nil
 }
 
+func (s *serviceImpl) GetLoadProcessBaseline(ctx context.Context, request *v1.GetProcessBaselineRequest) (*storage.ProcessBaseline, error) {
+	if err := validateKeyNotEmpty(request.GetKey()); err != nil {
+		return nil, errors.Wrap(errox.InvalidArgs, err.Error())
+	}
+	baseline, exists, err := s.dataStore.GetProcessBaseline(ctx, request.GetKey())
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		// Make sure the deployment still exists before we try to build a baseline.
+		_, exists, err := s.deployments.GetDeployment(ctx, request.GetKey().GetDeploymentId())
+		if err != nil {
+			return nil, err
+		}
+		if !exists {
+			return nil, errors.Wrapf(errox.NotFound, "deployment with id '%s' does not exist", request.GetKey().GetDeploymentId())
+		}
+
+		// Build an unlocked baseline
+		baseline, err = s.dataStore.BuildUnlockedProcessBaseline(ctx, request.GetKey())
```

Comment:
I believe that risk is mitigated by a couple of things.
1. I check to make sure the deployment exists just a few lines above this.  
2. If that were to happen and we create an orphaned baseline, it would get cleaned up the next time the pruning garbage collector job was executed.

If you feel we need greater protections, I'd be curious as to how to how we normally handle that so I could study that.